pkg/agent/nodeportlocal/portcache/port_table_windows_test.go

```diff
@@ -0,0 +1,180 @@
+		PortSearchStart: startPort,
+		PodPortRules:    mockPortRules,
+		LocalPortOpener: mockPortOpener,
+	}
+}
+
+func TestRestoreRules(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	podIP := "10.0.0.1"
+	nodePort1 := startPort
+	nodePort2 := startPort + 1
+	allNPLPorts := []rules.PodNodePort{
+		{
+			NodePort: nodePort1,
+			PodPort:  1001,
+			PodIP:    podIP,
+			Protocol: "tcp",
+		},
+		{
+			NodePort: nodePort1,
+			PodPort:  1001,
+			PodIP:    podIP,
+			Protocol: "udp",
+		},
+		{
+			NodePort: nodePort2,
+			PodPort:  1002,
+			PodIP:    podIP,
+			Protocol: "udp",
+		},
+	}
+
+	mockPortRules.EXPECT().AddRule(nodePort1, podIP, 1001, "tcp")
+	mockPortRules.EXPECT().AddRule(nodePort1, podIP, 1001, "udp")
+	mockPortRules.EXPECT().AddRule(nodePort2, podIP, 1002, "udp")
+
+	syncedCh := make(chan struct{})
+	err := portTable.RestoreRules(allNPLPorts, syncedCh)
+	require.NoError(t, err)
+}
+
+func TestDeleteRule(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	podIP := "10.0.0.1"
+	npData := &NodePortData{
+		NodePort: startPort,
+		PodIP:    "10.0.0.1",
+		PodPort:  1001,
+		Protocol: ProtocolSocketData{
+			Protocol: "tcp",
+		},
+	}
+
+	portTable.addPortTableCache(npData)
+	mockPortRules.EXPECT().DeleteRule(startPort, podIP, 1001, "tcp")
+	err := portTable.DeleteRule(podIP, 1001, "tcp")
+	require.NoError(t, err)
+}
+
+func TestDeleteRulesForPod(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	podIP := "10.0.0.1"
+	npData := &NodePortData{
+		NodePort: startPort,
+		PodIP:    "10.0.0.1",
+		PodPort:  1001,
+		Protocol: ProtocolSocketData{
+			Protocol: "tcp",
+		},
+	}
+
+	portTable.addPortTableCache(npData)
+	mockPortRules.EXPECT().DeleteRule(startPort, podIP, 1001, "tcp")
+	err := portTable.DeleteRulesForPod(podIP)
+	require.NoError(t, err)
+}
+
+func TestAddRule(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	tcs := []struct {
+		name        string
+		podIP       string
+		podPort     int
+		nodePort    int
+		protocol    string
+		expectedErr string
+	}{
+		{
+			name:     "valid entry for rule",
+			podIP:    "10.0.0.1",
+			podPort:  1001,
```

Comment:
I suppose these two fields can be removed if they are all the same in all cases. If they should change to cover more, then you may need to add more different cases.